src/group/managed_group/mod.rs

```diff
@@ -0,0 +1,276 @@
+        ciphersuite_name: CiphersuiteName,
+        key_package_bundle: KeyPackageBundle,
+        members: Vec<KeyPackage>,
+    ) -> Result<Self, GroupError> {
+        unimplemented!()
+    }
+
+    // === Membership management ===
+
+    /// Adds members to the group
+    pub fn add_members(
+        &mut self,
+        credential_bundle: &CredentialBundle,
+        aad: &[u8],
+        key_packages: Vec<KeyPackage>,
+    ) -> Result<Vec<MLSMessage>, GroupError> {
+        unimplemented!()
+    }
+
+    /// Removes members from the group
+    pub fn remove_members(
+        &mut self,
+        credential_bundle: &CredentialBundle,
+        aad: &[u8],
+        members: &[usize],
+    ) -> Result<Vec<MLSMessage>, GroupError> {
+        unimplemented!()
+    }
+
+    /// Creates proposals to add members to the group
+    pub fn propose_add_members(
+        &mut self,
+        credential_bundle: &CredentialBundle,
+        aad: &[u8],
+        key_packages: Vec<KeyPackage>,
+    ) -> Result<Vec<MLSMessage>, GroupError> {
+        unimplemented!()
+    }
+
+    /// Creates proposals to remove members from the group
+    pub fn propose_remove_members(
+        &mut self,
+        credential_bundle: &CredentialBundle,
+        aad: &[u8],
+        members: &[usize],
+    ) -> Result<Vec<MLSMessage>, GroupError> {
+        unimplemented!()
+    }
+
+    /// Gets the current list of members
+    pub fn get_members(&self) -> &[Credential] {
+        unimplemented!()
+    }
+
+    // === Process messages ===
+
+    /// Processes any incoming message from the DS (MLSPlaintext & MLSCiphertext) and triggers the corresponding callback functions
+    pub fn process_message(&mut self, message: MLSMessage) {
+        unimplemented!()
+    }
+
+    // === Create application messages ===
+
+    /// Create an application message
+    pub fn create_message(
+        &mut self,
+        credential_bundle: &CredentialBundle,
+        aad: &[u8],
+        message: &[u8],
+    ) {
+        unimplemented!()
+    }
+
+    // === Export secrets ===
+
+    /// Exports a secret from the current epoch
+    pub fn export_secret(&self) -> Vec<u8> {
+        unimplemented!()
+    }
+
+    // === Advanced functions ===
+
+    /// Updates the own leaf node
+    pub fn self_update(
+        &mut self,
+        credential_bundle: &CredentialBundle,
+    ) -> Result<Vec<MLSMessage>, GroupError> {
+        unimplemented!()
+    }
+
+    /// Creates a proposal to update the own leaf node
+    pub fn propose_self_update(
+        &mut self,
+        credential_bundle: &CredentialBundle,
+    ) -> Result<Vec<MLSMessage>, GroupError> {
+        unimplemented!()
+    }
+
+    /// Returns a list of proposal
+    pub fn get_pending_proposals(&self) -> Vec<Proposal> {
+        unimplemented!()
+    }
+
+    // === Load & save (WIP) ===
+
+    /// Loads the state from persisted state
+    pub fn load(state: &[u8], managed_group_config: &ManagedGroupConfig) -> ManagedGroup {
```

Comment:
The state could get so big that we don't want to have it in a slice/vector. So this should probably either get a `dyn Read` or a path to read from or something similar.

Reply:
Absolutely, this is a total placeholder and should be replaced with whatever we'll use for persistence.